Include/internal/pycore_tagged.h

```diff
@@ -0,0 +1,194 @@
+#ifndef Py_INTERNAL_TAGGED_H
+#define Py_INTERNAL_TAGGED_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+#include <stddef.h>
+
+// Mark an object as supporting deferred reference counting. This is a no-op
+// in the default (with GIL) build. Objects that use deferred reference
+// counting should be tracked by the GC so that they are eventually collected.
+extern void _PyObject_SetDeferredRefcount(PyObject *op);
+
+static inline int
+_PyObject_HasDeferredRefcount(PyObject *op)
+{
+#ifdef Py_GIL_DISABLED
+    return (op->ob_gc_bits & _PyGC_BITS_DEFERRED) != 0;
+#else
+    return 0;
+#endif
+}
```

Comment:
I tried this in the original branch but we can't easily for some reason. We can't `#include pycore_object.h` in `pycore_tagged.h` because it causes anything that includes `pycore_interp.h` to fail to compile. Not sure why.

Reply:
Ah, my old enemy...circular includes.  When this has happened to me in the past, I solved this by splitting out the problematic parts of the header file into a new "pycore_..._state.h".  It holds just the structs needed by pycore_interp.h.  Then pycore_interp.h includes only the "_state.h" file.  You can see examples in Include/internal.